### PR TITLE
[avro-schema] fix: enum logic (sanitize, remove collisions)

### PR DESCRIPTION
**Pull request by joscha**
Currently, enums in Avro schemas are generated directly from the model, whether they are valid or not.

Unfortunately, Avro schema values [follow tighter rules](https://avro.apache.org/docs/1.12.0/specification/#enums).

This pull request sanitizes enum values from the Swagger spec to follow the Avro schema by:
* replacing invalid characters with `_`
* prefixing values starting with numbers with `_`
* removing collisions that happen because of the above or otherwise by adding a numbered suffix.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. - @sgadouar?

### CODE REVIEW

File: samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "fields": [
+    {
+      "name": "type",
+      "type": ["null", {
+        "type": "enum",
+        "name": "Sample_type",
+        "symbols": [
+          "a",
+          "b",
+          "uh_oh",
```
Comment (joscha) [inline samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc:15]: `-` to `_`

File: samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          "_0h_oh",
```
Comment (joscha) [inline samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc:16]: prefixed with `_`

File: samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          "coll_ision",
+          "coll_ision2"
```
Comment (joscha) [inline samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc:18]: count suffix

**Comment by wing328** [comment]
thanks for the PR.

cc @sgadouar

File: samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (joscha) [inline samples/openapi3/schema/valid-enums/avro-schema-enum/Sample.avsc:1]: happy to express this at unit tests instead if you prefer @sgadouar

**Pull request merged**